Added wallet and device schema(table). Added device inside the admin …

eoliveros opened the pull request:
…tab with filters.

eoliveros committed:
Added wallet and device schema(table). Added device inside the admin tab with filters.

djpnewton commented on zapd/models.py:
```diff
@@ -161,6 +161,32 @@ def in_status(cls, session, status):
 
+class Wallet(db.Model):
+    id = db.Column(db.Integer(), primary_key=True)
+    wallet_address = db.Column(db.String(), unique=True)
```
this can just be 'address' (because the table name is 'wallet')

eoliveros replied:
changed.

djpnewton commented on zapd/models.py:
```diff
@@ -161,6 +161,32 @@ def in_status(cls, session, status):
+
+class Device(db.Model):
+    id = db.Column(db.Integer(), primary_key=True)
+    device_imei_id = db.Column(db.String(), unique=True)
```
_id is probably redundant in device_imei_id

eoliveros replied:
changed.

djpnewton commented on zapd/models.py:
```diff
@@ -161,6 +161,32 @@ def in_status(cls, session, status):
+    id = db.Column(db.Integer(), primary_key=True)
+    device_imei_id = db.Column(db.String(), unique=True)
+    device_os = db.Column(db.String())
+    device_mobile_no = db.Column(db.String())
```
lets add device_details (maybe we can add other stuff, make and model etc)
also add date field

djpnewton replied:
ok I am looking at https://github.com/react-native-community/react-native-device-info to see what info we can get (and want)

so far i have:
  app version
  os, os version
  manufacturer, brand, deviceId/model

So we can add these fields and remove imei and mobile number (only available on android)

djpnewton commented:
I think we should prefix these tables with something... App Metrics? (AMWallet, AMDevice?)

eoliveros commented:
> I think we should prefix these tables with something... App Metrics? (AMWallet, AMDevice?)

Changing the class Name: ie(AMWallet, AMDevice)
So the classes would look like:
class AMWallet

class AMDevice

like the above?

djpnewton commented:
yes i want to make it clear that these are kinda separate from the main app

djpnewton committed:
- fix amdevice schema to match what zap app produces
- fix amdevice/amwallet relationship
- add api to create amwallet/amdevice records

djpnewton reviewed (commented):
added commit to address issues, i think it just needs testing now

djpnewton commented on zapd/admin.py:
```diff
@@ -18,6 +18,7 @@
 admin.add_view(UserModelView(User, db.session, category='Admin'))
 admin.add_view(RestrictedModelView(Role, db.session, category='Admin'))
 admin.add_view(RestrictedModelView(Category, db.session, category='Admin'))
+admin.add_view(AMDeviceRestrictedModelView(AMDevice, db.session, category='Admin'))
```
We should change this from device table to wallet table, we will want to see the different wallet addresses and filter for any with more then one device

eoliveros replied:
Trying to pull the latest update from the upstream, i cant seem to pull it.

I manually copied the changes and placed it in my repo. Will look in to the other changes mentioned above.

Been trying to do this in my local repo shoudl have pulled the remote repo first.

djpnewton replied:
I committed directly to your branch: https://github.com/eoliveros/zapd/commit/94691b9f597d785f5bb36fe91d966453b3fbdb9f

eoliveros replied:
after pointing to AMWallet:
![image](https://user-images.githubusercontent.com/47799652/83822441-24211d80-a725-11ea-8cac-e2724be75d28.png)


Obviously, now we need to add the other columns from the other table.

eoliveros replied:
changed the schema:
```
CREATE TABLE amdevice (
        id INTEGER NOT NULL,
        date DATETIME,
        app_version VARCHAR,
        os VARCHAR,
        os_version VARCHAR,
        manufacturer VARCHAR,
        brand VARCHAR,
        device_id VARCHAR,
        PRIMARY KEY (id)
);
CREATE TABLE amwallet (
        id INTEGER NOT NULL,
        address VARCHAR,
        amdevice_id INTEGER,
        PRIMARY KEY (id),
        FOREIGN KEY(amdevice_id) REFERENCES amdevice (id)
);
```

The amwallet.address used to be unique, but when I'm inserting the records to make it look like there's duplicate, it would fail because of "unique constraint".


The amdevice(will disable later) screen currently looks like:
![image](https://user-images.githubusercontent.com/47799652/83982278-abff6580-a979-11ea-92eb-dae15e68e8ea.png)

The amwallet looks like:
![image](https://user-images.githubusercontent.com/47799652/83982303-e5d06c00-a979-11ea-9df8-49a103209b7c.png)

will look at creating the filter.

eoliveros replied:
Resolving this since "AMDeviceRestrictedModelView" is not in the admin.py.

djpnewton commented on zapd/models.py:
```diff
@@ -310,6 +353,27 @@ def get_options(self, view):
 def get_status_options():
     return [(proposal.status, proposal.status) for proposal in Proposal.query.distinct(Proposal.status).group_by(Proposal.status)]
 
+class FilterByDeviceOS(BaseSQLAFilter):
```
dont need this new block of filters

eoliveros replied:
no longer there

djpnewton commented on zapd/models.py:
```diff
@@ -759,3 +823,34 @@ def is_accessible(self):
+            self.can_export = True
+            return True
+
+class AMDeviceRestrictedModelView(sqla.ModelView):
```
dont need this

eoliveros replied:
using this as a test at the moment. Want to see what the other side looks like. will disable soon.

eoliveros replied:
This is no longer there from the latest update i pulled. Resolving.

djpnewton commented on zapd/models.py:
```diff
@@ -759,3 +823,34 @@ def is_accessible(self):
             self.can_export = True
             return True
 
+class WalletRestrictedModelView(sqla.ModelView):
```
update this with a filter to search for wallets with more then one device

and make it visible via the flask admin UI

eoliveros replied:
just pulled the latest update.
Erroring at the moment when I try to go to the AM Wallet page:

Its refering to the amdevice_list.html

Currently this page doesn't appear to exists.

Not sure why, but there is a line in the AMWalletRestrictedModelView:

list_template = 'amdevice_list.html'

Which forces the page to go to  amdevice_list.html, which currently doesnt exists. Commenting this line to get the page up.

djpnewton committed:
fix flask admin side (table to show, filters)
fix am wallet api
